Author: Daniel Leidert <dleidert@debian.org>
Description: Fix building with bison 3.x. YYLEX_PARAMS is deprecated. The
 syntax needs a fix otherwise the package will FTBFS.
Bug-Debian: http://bugs.debian.org/733359

--- a/Code/GraphMol/SmilesParse/smarts.yy
+++ b/Code/GraphMol/SmilesParse/smarts.yy
@@ -1,5 +1,3 @@
-%pure_parser
-
 %{
 
   // $Id$
@@ -23,7 +21,6 @@
 
 
 #define YYDEBUG 1
-#define YYLEX_PARAM scanner
 
 void
 yysmarts_error( const char *input,
@@ -46,6 +43,8 @@
 }
 %}
  
+%define api.pure
+%lex-param   {yyscan_t *scanner}
 %parse-param {const char *input}
 %parse-param {std::vector<RDKit::RWMol *> *molList}
 %parse-param {void *scanner}
--- a/Code/GraphMol/SmilesParse/smiles.yy
+++ b/Code/GraphMol/SmilesParse/smiles.yy
@@ -1,5 +1,3 @@
-%pure_parser
-
 %{
 
   // $Id$
@@ -23,7 +21,6 @@
 extern int yysmiles_lex(YYSTYPE *,void *);
 
 #define YYDEBUG 1
-#define YYLEX_PARAM scanner
 
 void
 yysmiles_error( const char *input,
@@ -48,6 +45,8 @@
 
 %}
 
+%define api.pure
+%lex-param   {yyscan_t *scanner}
 %parse-param {const char *input}
 %parse-param {std::vector<RDKit::RWMol *> *molList}
 %parse-param {std::list<unsigned int> *branchPoints}
--- a/Code/GraphMol/SLNParse/sln.yy
+++ b/Code/GraphMol/SLNParse/sln.yy
@@ -1,5 +1,3 @@
-%pure_parser
-
 %{
 
   // $Id$
@@ -77,14 +75,14 @@
 
 
 %}
- 
+
+%define api.pure
+%lex-param   {yyscan_t *scanner}
 %parse-param {const char *input}
 %parse-param {std::vector<RDKit::RWMol *> *molList}
 %parse-param {bool doQueries}
 %parse-param {void *scanner}
 
-%lex-param {yyscan_t *scanner}
-
 %union {
   int                      mol_T;
   RDKit::Atom *            atom_T;
